humanoid_league_team_communication/src/humanoid_league_team_communication/humanoid_league_team_communication.py

```diff
@@ -0,0 +1,401 @@
+#!/usr/bin/env python3
+import math
+import socket
+from typing import Optional
+
+import rospy
+import rospkg
+import struct
+
+import tf2_ros
+import transforms3d
+from geometry_msgs.msg import Twist, PoseWithCovarianceStamped
+from humanoid_league_msgs.msg import GameState, PoseWithCertaintyArray, TeamData, ObstacleRelativeArray, \
+    ObstacleRelative, Strategy
+from tf2_geometry_msgs import PointStamped, PoseStamped
+
+import robocup_extension_pb2
+
+
+class HumanoidLeagueTeamCommunication:
+    def __init__(self):
+        rospack = rospkg.RosPack()
+        self._package_path = rospack.get_path("humanoid_league_team_communication")
+
+        rospy.init_node("humanoid_league_team_communication")
+        rospy.loginfo("Initializing humanoid_league_team_communication...", logger_name="team_comm")
+
+        self.player_id = rospy.get_param('bot_id')
+        self.team_id = rospy.get_param('team_id')
+
+        self.config = rospy.get_param("~")
+
+        self.target_host = self.config['target_host']
+        if self.target_host == '127.0.0.1':
+            # local mode, bind to port depending on bot id
+            self.target_ports = self.config['local_target_ports']
```

Comment:
What do you mean?

Reply:
In local mode, the `receive_port` (depending on the `player_id`) is also in the `target_ports` list.
So it will try to send to itself? Maybe this can cause problems, either way it is unnecessary.